pkg/config/url.go

```diff
@@ -10,20 +10,24 @@ import (
 
 // GetPorts returns the ports stored in the config for the component
 // returns default i.e nil if nil
-func (lc *LocalConfig) GetPorts() ([]string, error) {
+func (lc *LocalConfig) GetComponentPorts() ([]string, error) {
 	if lc.componentSettings.Ports == nil {
 		return nil, nil
 	}
 	return *lc.componentSettings.Ports, nil
 }
 
+func (lc *LocalConfig) GetContainerPorts(container string) ([]string, error) {
+	return nil, fmt.Errorf("Not implemented for local config")
```

Comment:
I think we can call GetComponentPorts instead of the error here as there is only one container in case of s2i components.

Reply:
I think it makes sense that function handles only what it is intended for. I would expect `GetContainerPorts` only to get ports for the specified containers and if no container being specified is an error state in my book

If you check the localconfig changes for s2i. It only calls `GetComponentPorts` directly and never calls `GetContainerPorts`

Reply:
> If you check the localconfig changes for s2i. It only calls GetComponentPorts directly and never calls GetContainerPorts

OK. Please change the error message to start with a lower case.

Reply:
fixed